chips/apollo3/src/lib.rs

```diff
@@ -104,7 +104,7 @@ pub unsafe fn init() {
     scb::disable_fpca();
 
     // This ensures the FPU is actually disabled
-    llvm_asm!("svc 0xff" : : : "r0","r1","r2","r3","r12" : "volatile" );
+    asm!("svc 0xff", out("r0") _, out("r1") _, out("r2") _, out("r3") _, out("r12") _);
```

Comment:
```suggestion
    asm!("svc 0xff", out("r0") _, out("r1") _, out("r2") _, out("r3") _, out("r12") _, options(preserves_flags));
```

Reply:
Isn't this going to run the svc handler? I am confused why this doesn't cause us to cross the userspace/kernel boundary when this is exactly what we use to cross the boundary in `switch_to_user`.

Reply:
I guess that is really a question for @alistair23

Reply:
Yes, it should call `svc_handler()`.

`svc_handler()` can also trap back to the kernel, which is what happens in this case.

I'm not really sure why we need this here, but I remember that it didn't work otherwise.